### DISCUSSION TITLE
Please update the javadoc to clearly indicate which classes are thread-safe and which are not. This will aid performance optimization for code using the library.

### DISCUSSION THREAD

**Discussion by petesramka** [networknt/json-schema-validator#1044] [Ideas]
One of your big selling points is that your library is fast. Without knowing which classes are thread-safe, I don't know which objects I can share across threads. Knowing which objects can be shared across threads determines which types of caches implementations can be used. This, in turn, allows code using your library to be performance optimized.

**Comment by justin-tay** [comment]
You can read the [Performance Considerations](https://github.com/networknt/json-schema-validator?tab=readme-ov-file#performance-considerations) section of the readme.

You would need to be more specific on which classes you need clarification of as you make it sound like you expect every single class to indicate whether it is thread safe or not. Other than instances of `JsonSchema` and `JsonSchemaFactory`, which in general should be thread-safe given their configurations are not changed after being constructed, I don't know what other classes you would need to know.

**Reply by petesramka** [reply]
Please don't take this personally, but just the fact that you list some classes as "thread-safe given their configurations are not changed after being constructed" and are surprised that someone would "expect every single class to indicate whether it is thread safe or not" indicates a lack of understanding of Java concurrency. But this lack of knowledge is very common amongst Java developers, so you are not alone. If a class is only thread-safe under certain conditions, then it is not thread-safe. Also, it is considered a best practice to document the thread-safety of all classes. There is a really easy way to do this with annotations:
https://jcip.net/annotations/doc/

**Reply by justin-tay** [reply]
I somewhat expected this comment. You can then assume nothing is threadsafe then as the factory holds an instance of Jackson's ObjectMapper which has the same caveat on threadsafety regarding the configuration.
